libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/DefaultTracer.java

```diff
@@ -76,4 +77,9 @@ protected void addDefaultAttributes(Span span) {
         span.addAttribute(THREAD_NAME, Thread.currentThread().getName());
     }
 
+    @Override
+    public Releasable newTracerContextStorage() {
+        return tracerContextStorage.newTracerContextStorage();
+    }
+
```

Comment:
Looks non-intuitive, since this class already has a `close` method the ctor of which instantiates the tracer context storage but doesn't worry about releasing it. Would prefer ctor create the context storage and close to tear dereference or release

Reply:
@Bukhtawar Thanks for reviewing. This method is basically needed to fork out the new storage temporarily and should be able to restore the context once the temporary work is over. Example mentioned here https://github.com/opensearch-project/OpenSearch/issues/8428#issue-1787742599. So, now this temporary context needs to be closed/released externally by the caller. Otherwise the context storage need not to be closed/released as per the current implementations but can be enhanced. Please let me know If I am missing something here.